registration/models.py

```diff
@@ -422,7 +422,7 @@ def _activate(self, profile, get_profile):
         """
 
         if not profile.user.is_active and not profile.activated:
-            site = Site.objects.get_current()
+            site = get_current_site()
```

Comment:
Can I try to change `RegistrationManager.activate_user` interface to `activate_user(self, site, activation_key, get_profile=False)` like `RegistrationManager.create_inactive_user(self, site, new_user=None, send_email=True, request=None, profile_info={}, **user_info)`